catalog/spatial/registry/registry-api-impl/src/main/java/org/codice/ddf/registry/api/impl/RegistryStorePublisher.java

```diff
@@ -88,79 +114,69 @@ public void setUnavailable() {
-                String localRegistryId = registryIdentityMetacard.getAttribute(
-                        RegistryObjectMetacardType.REGISTRY_ID)
-                        .getValue()
-                        .toString();
-                Security.runAsAdminWithException(() -> {
-                    registryPublicationService.publish(localRegistryId,
-                            registryStore.getRegistryId());
-                    return null;
-                });
+            StringBuilder logMessage = new StringBuilder(
+                    "RegistryStore missing id. Unable to complete ");
+            if (!publish) {
+                logMessage.append("un");
             }
-        } catch (Exception e) {
-            LOGGER.error("Failed to publish registry configuration to {}",
-                    ((RegistryStoreImpl) registryStore).getRemoteName());
-        }
-    }
-
-    void registryUnpublish(RegistryStore registryStore) {
-
-        if (registryStore.getRegistryId()
-                .isEmpty()) {
+            logMessage.append("publish request.");
+            LOGGER.warn(logMessage.toString());
             return;
         }
-        try {
 
-            Optional<Metacard> registryIdentityMetacardOpt =
-                    Security.runAsAdminWithException(() -> federationAdminService.getLocalRegistryIdentityMetacard());
-
-            if (registryIdentityMetacardOpt.isPresent()) {
-                Metacard registryIdentityMetacard = registryIdentityMetacardOpt.get();
-                String localRegistryId = registryIdentityMetacard.getAttribute(
-                        RegistryObjectMetacardType.REGISTRY_ID)
-                        .getValue()
-                        .toString();
-                Security.runAsAdminWithException(() -> {
-                    registryPublicationService.unpublish(localRegistryId,
-                            registryStore.getRegistryId());
-                    return null;
-                });
-            }
-        } catch (Exception e) {
-            LOGGER.error("Failed to unpublish registry configuration to ",
-                    ((RegistryStoreImpl) registryStore).getRemoteName());
-        }
+        executor.schedule(() -> {
+            try {
+                Optional<Metacard> registryIdentityMetacardOpt =
+                        Security.runAsAdminWithException(() -> federationAdminService.getLocalRegistryIdentityMetacard());
+
+                if (registryIdentityMetacardOpt.isPresent()) {
+                    Metacard registryIdentityMetacard = registryIdentityMetacardOpt.get();
+                    String localRegistryId = registryIdentityMetacard.getAttribute(
+                            RegistryObjectMetacardType.REGISTRY_ID)
+                            .getValue()
+                            .toString();
+                    Security.runAsAdminWithException(() -> {
+                        if (publish) {
```

Comment:
I broke out the inner lambda to pass in when the method is called, but it ended up getting a bit messier. The initial problem was the inner lambda requires a variable that is derived in the lambda above it. I tried extrapolating some of the logic into another method, but it actually seems pretty nice to have the security calls and lambdas bundled up into a single nice package that all operates on this thread. I still used the "publish"/"unpublish" idea from your comment, but I still have one inner conditional check. Please let me know if I missed the target on this one.